cassovary-core/src/main/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraph.scala

```diff
@@ -115,7 +118,8 @@ case class BipartiteSide(nodes: Array[BipartiteNode], numOfNodes: Int, numOfOutE
 class BipartiteGraph(val leftNodes: Array[BipartiteNode], val leftNodeCount: Int,
                      val leftOutEdgeCount: Long, val rightNodes: Array[BipartiteNode],
                      val rightNodeCount: Int, val rightOutEdgeCount: Long,
-                     val bipartiteGraphDir: BipartiteGraphDir.BipartiteGraphDir) extends Graph[BipartiteNode] {
+                     val bipartiteGraphDir: BipartiteGraphDir.BipartiteGraphDir,
+                     val isLeftNode: (Int => Option[Boolean]) = (_ => None)) extends Graph[BipartiteNode] {
```

Comment:
if leftNode and rightNodes are explicitly specified, the function isLeftNode is not useful to also specify. So I would not have this extra isLeftNode parameter.

Adding 'unique' here in the constructor of the graph makes sense -- let's call it nodeIdsUnique (please add a comment about what this means).